sfxclient/sfxclient.go

```diff
@@ -283,6 +271,81 @@ func (s *Scheduler) RemoveGroupedCallback(group string, db Collector) {
 	}
 }
 
+// CollectorAggregator is an
+type CollectorAggregator interface {
+	Var() expvar.Var
+	CollectDatapoints() []*datapoint.Datapoint
+	AddCallback(db Collector)
+	DefaultDimensions(dims map[string]string)
+	GroupedDefaultDimensions(group string, dims map[string]string)
+	AddGroupedCallback(group string, db Collector)
+	RemoveCallback(db Collector)
+	RemoveGroupedCallback(group string, db Collector)
+}
+
+// DefaultHTTPErrorHandler is the default way to handle errors by a CollectorHandler.  It attempts to reply with a 500 http error
+var DefaultHTTPErrorHandler = func(err error, w http.ResponseWriter, req *http.Request) error {
+	http.Error(w, err.Error(), http.StatusInternalServerError)
+	return err
+}
+
+// CollectorHandler aggregates collectors handles http requests for datapoints
+type CollectorHandler struct {
+	*CollectorCallbackManager
+	HTTPErrorHandler   func(error, http.ResponseWriter, *http.Request) error
+	jsonMarshallerFunc func(v interface{}) ([]byte, error)
+}
+
+// NewCollectorHandler gets you the new CollectorHandler
+func NewCollectorHandler() *CollectorHandler {
+	return &CollectorHandler{
+		CollectorCallbackManager: NewCollectorCallbackManager(),
+		HTTPErrorHandler:         DefaultHTTPErrorHandler,
+		jsonMarshallerFunc:       json.Marshal,
+	}
+}
+
+// Datapoints exposes a handler func to return datapoints from all callbacks
+func (c *CollectorHandler) Datapoints(w http.ResponseWriter, req *http.Request) {
```

Comment:
not sure "Datapoints" is a good name here.  Maybe thing about how this would compose with zippers and if there's a better signature for that or is the correct thing to just write a closure func(v interface{}) ([]byte, error) that does it.

Reply:
I can rename it, but the Method signature func(http.ResponseWriter, *http.Request) is a requirement to be a handler function.  The idea was that this request handler implementation could be wired up to a server.  If I switch to func(v interface{}), then I have to wrap it inside of a func(http.ResponseWriter, *http.Request) where ever I use it.